extensions/iframe/src/iframe.js

```diff
@@ -0,0 +1,53 @@
+let call;
+
+const props = {};
+
+/* Extension configuration */
+
+function configure(config) {
+  Object.assign(props, config);
+}
+
+function handleIframeButton() {
+  const s = call.iframeState();
+  if (s.open === true) {
+    call.closeIframe();
+  } else {
+    let url = prompt("Enter a URL", props.url);
+    if (url) {
+      call.openIframe(url);
+    }
+  }
+}
+
+function beforeCreateFrame(parentEl, properties) {
+  //TODO: can we avoid this somehow?
+  if (!properties.customTrayButtons) {
+    properties.customTrayButtons = {};
+  }
+  properties.customTrayButtons.iframe = {
+    iconPath: "https://www.svgrepo.com/show/107601/globe.svg",
+    label: "Iframe",
+    tooltip: "Open Iframe",
+  };
+
+  return [parentEl, properties];
+}
+
+function afterCreateFrame(c) {
+  call = c;
+  //TODO: Can we avoid this somehow?
+  call.on("custom-button-click", (e) => {
+    switch (e.button_id) {
+      case "iframe":
+        handleIframeButton();
```

Comment:
Instead of making `call` global here I'd pass it into handleIFrameButton
```
handleIframeButton(c)
```

It adds extra parameters but it's a lot easier to reason about local state inside a function then to worry about a variable globally that may or may not be affected by another function